flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/physical/stream/StreamPhysicalMLPredictTableFunctionRule.java

```diff
@@ -0,0 +1,93 @@
+ */
+
+package org.apache.flink.table.planner.plan.nodes.physical.stream;
+
+import org.apache.flink.table.ml.PredictRuntimeProvider;
+import org.apache.flink.table.planner.calcite.RexModelCall;
+import org.apache.flink.table.planner.functions.sql.ml.SqlMLPredictTableFunction;
+import org.apache.flink.table.planner.plan.nodes.FlinkConventions;
+import org.apache.flink.table.planner.plan.nodes.logical.FlinkLogicalTableFunctionScan;
+import org.apache.flink.table.planner.plan.trait.FlinkRelDistribution;
+
+import org.apache.calcite.plan.RelOptRule;
+import org.apache.calcite.plan.RelOptRuleCall;
+import org.apache.calcite.plan.RelTraitSet;
+import org.apache.calcite.rel.RelNode;
+import org.apache.calcite.rel.convert.ConverterRule;
+import org.apache.calcite.rex.RexCall;
+import org.checkerframework.checker.nullness.qual.Nullable;
+
+/**
+ * Rule to convert a {@link FlinkLogicalTableFunctionScan} with ml_predict call into a {@link
+ * StreamPhysicalMLPredictTableFunction}.
+ */
+public class StreamPhysicalMLPredictTableFunctionRule extends ConverterRule {
+
+    public static final StreamPhysicalMLPredictTableFunctionRule INSTANCE =
+            new StreamPhysicalMLPredictTableFunctionRule(
+                    Config.INSTANCE.withConversion(
+                            FlinkLogicalTableFunctionScan.class,
+                            FlinkConventions.LOGICAL(),
+                            FlinkConventions.STREAM_PHYSICAL(),
+                            "StreamPhysicalModelTableFunctionRule"));
+
+    private StreamPhysicalMLPredictTableFunctionRule(Config config) {
+        super(config);
+    }
+
+    @Override
+    public boolean matches(RelOptRuleCall call) {
+        final FlinkLogicalTableFunctionScan scan = call.rel(0);
+        final RexCall rexCall = (RexCall) scan.getCall();
+        if (!(rexCall.getOperator() instanceof SqlMLPredictTableFunction)) {
+            return false;
+        }
+
+        final RexModelCall modelCall = (RexModelCall) rexCall.getOperands().get(1);
+        return modelCall.getModelProvider() instanceof PredictRuntimeProvider;
+    }
+
+    @Override
+    public @Nullable RelNode convert(RelNode rel) {
+        final FlinkLogicalTableFunctionScan scan = (FlinkLogicalTableFunctionScan) rel;
+        final RelNode newInput = applyDistributionOnInput(scan.getInput(0));
+
+        final RelTraitSet providedTraitSet =
+                rel.getTraitSet().replace(FlinkConventions.STREAM_PHYSICAL());
+
+        // TODO:
+        // Get model provider and context resolved model from RexModelCall
+        // Get table input from descriptor
+        // Get config from map
+        // Create ModelProviderSpec similar to DynamicTableSourceSpec and TemporalTableSourceSpec
+        return new StreamPhysicalMLPredictTableFunction(
+                scan.getCluster(), providedTraitSet, newInput, scan, scan.getRowType());
+    }
+
+    private static RelNode applyDistributionOnInput(RelNode input) {
```

Comment:
Why we need to apply distribution? Can we just call ` RelOptRule.convert(scan.getInput(0), FlinkConventions.STREAM_PHYSICAL)` directly?

Reply:
I looked at https://github.com/apache/flink/blob/master/flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/physical/stream/StreamPhysicalWindowTableFunctionRule.scala#L49-L55 and https://github.com/apache/flink/blob/master/flink-table/flink-table-planner/src/main/java/org/apache/flink/table/planner/plan/nodes/physical/stream/StreamPhysicalProcessTableFunctionRule.java#L111-L122, looks ml_predict shouldn't care about the `TableCharacteristic`. So I put default here.

Reply:
I mean we can just use the RelOptRule.convert(scan.getInput(0), FlinkConventions.STREAM_PHYSICAL) to replace all codes in the `applyDistributionOnInput`